interfaces/cython/cantera/thermo.pyx

```diff
@@ -1650,6 +1650,59 @@ cdef class ThermoPhase(_SolutionBase):
             cdef CxxUnits units = self.thermo.standardConcentrationUnits()
             return Units.copy(units)
 
+    # methods for plasma
+    def set_electron_energy_distribution(self, levels, distribution):
+        """
+        Set electron energy distribution. When this method is used, electron
+        temperature is calculated from the distribution.
+
+        :param levels:
+            vector of electron energy levels [eV]
+        :param distribution:
+            vector of distribution
+        """
+        if self.thermo_model != "Plasma":
+            raise TypeError("Only valid for PlasmaPhase (thermo_model='Plasma').")
```

Comment:
Can I use <CxxPlasmaPhase*>self.thermo? Or I need to use ctypedef CxxPlasmaPhase* CxxPlasmaPhasePtr? I am not sure where I should put it.

Reply:
`<CxxPlasmaPhase*>self.thermo` is an unchecked cast that just blindly treats `self.thermo` as a `CxxPlasmaPhase*`. This is why you need to use `dynamic_cast` -- it will return an null pointer if the object referred to by `self.thermo` is not actually of the correct type.

The reason for the `ctypedef` is just a workaround syntax limitation of Cython, in that you can't write `dynamic_cast[CxxPlasmaPhase*](self.thermo)`. The `ctypedef` should go in the global scope at the top of `thermo.pyx`, probably either just before or just after `cdef enum ThermoBasisType`.